pkg/tenant/topics.go

```diff
@@ -257,30 +262,137 @@ func (s *Server) TopicUpdate(c *gin.Context) {
 	c.JSON(http.StatusOK, t.ToAPI())
 }
 
-// TopicDelete deletes a topic from a user's request with a given ID
-// and returns a 200 OK response instead of an error response.
+// TopicDelete completely destroys a topic, removing the metadata in Trtl and as well
+// as all of the data in Ensign. Because this is irreversible, the first call returns
+// a confirmation token to the user. The user must provide this token in a subsequent
+// request in order to confirm the deletion. Because this operation is asynchronous,
+// the endpoint returns a 202 Accepted response.
 //
 // Route: /topic/:topicID
 func (s *Server) TopicDelete(c *gin.Context) {
 	var (
-		err error
+		err    error
+		ctx    context.Context
+		claims *tokens.Claims
 	)
 
+	// User credentials are required for Quarterdeck requests
+	if ctx, err = middleware.ContextFromRequest(c); err != nil {
+		log.Error().Err(err).Msg("could not create user context from request")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch credentials for authenticated user"))
+		return
+	}
+
+	// User claims are required to verify that the user owns the topic
+	if claims, err = middleware.GetClaims(c); err != nil {
+		log.Error().Err(err).Msg("could not fetch claims from context")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch claims from context"))
+		return
+	}
+
 	// Get the topic ID from the URL and return a 400 response
-	// if the topic does not exist.
+	// if the ID is not parseable
 	var topicID ulid.ULID
 	if topicID, err = ulid.Parse(c.Param("topicID")); err != nil {
-		log.Error().Err(err).Msg("could not parse topic ulid")
-		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse topic ulid"))
+		log.Warn().Err(err).Msg("could not parse topic id")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse topic id"))
```

Comment:
I'm thinking I'd actually prefer to return a 404 here (I know this has nothing to do with your PR, but just thinking about it again. E.g. if the user puts in `/project/:projectID/topics/foo` - that's not parseable as a ULID, but it should be not found not 400 right? The 404 is a little more secure and will allow us to change ID types in the future. 

```suggestion
		c.JSON(http.StatusNotFound, api.ErrorResponse("topic not found"))
```

Reply:
Agreed, I think we've gone back and forth but it does seem to make a bit more sense to return 404 since the ID is a part of the URL.